micro-rdk/src/common/ota.rs

```diff
@@ -414,84 +427,98 @@ impl<S: OtaMetadataStorage> OtaService<S> {
         let mut got_info = false;
 
         log::info!("writing new firmware to address `{:#x}`", self.address,);
+        let mut stream = response.into_data_stream();
+
+        loop {
+            match stream
+                .try_next()
+                .map_err(|e| FrameError::Network(e))
+                .or(async {
+                    async_io::Timer::after(Duration::from_secs(30)).await;
+                    Err(FrameError::Timeout(30))
```

Comment:
- can make this a `const usize`
- the goal is to timeout if a single frame is unresponsive for more than 30seconds, unblocking the caller (ConfigMonitor) so it can try again in 10 seconds (ConfigMonitor's default period)